nearest.go

```diff
@@ -0,0 +1,52 @@
+package osrm
+
+import geo "github.com/paulmach/go.geo"
+
+// NearestRequest represents a request to the nearest method
+type NearestRequest struct {
+	Profile  string
+	GeoPath  GeoPath
+	Bearings []Bearing
+
+	Number int
+}
+
+// NearestResponse represents a response from the nearest method
+type NearestResponse struct {
+	Waypoints []NearestWaypoint `json:"waypoints"`
+}
+
+// NearestWaypoint represents a nearest point on a nearest query
+type NearestWaypoint struct {
+	Location geo.Point `json:"location"`
+	Distance float64   `json:"distance"`
+	Name     string    `json:"name"`
+	Hint     string    `json:"hint"`
+	Nodes    []uint64  `json:"nodes"`
+}
+
+type nearestResponseOrError struct {
+	responseStatus
+	NearestResponse
+}
+
+func (r NearestRequest) request() *request {
+	opts := options{}
+	opts.addInt("number", r.Number)
```

Comment:
Let's not make this param required - OSRM has default value here (`number integer >= 1 (default 1)`), and zero seems like not a valid input.

So could you please add zero value check like `if r.Number != 0 { opts.addInt(..) }`?

Reply:
done `if r.Number >= 1`